### ISSUE TITLE
send completed event

### ISSUE THREAD

**Issue by Alshathry** [flightcrank/JRatioBoost#10]
can you please implement completed event

**Comment by flightcrank** [comment]
what problem are you having ? its not really needed

**Comment by Alshathry** [comment]
when there is no completed event, it easy to tracker to know you are cheating i guess

**Comment by flightcrank** [comment]
>  uploads kb only works with arrows

dont know what you mean by that.

>when there is no completed event, it easy to tracker to know you are cheating i guess.

well not really. trackers allow cross seeding where you have already completed a torrent from another tracker and just let you seed their tracker. never triggering the completed event.

**Comment by Alshathry** [comment]
upload kb mount only works with arrows not increasing when typing desired amount

if i want to increase the ratio on random torrent not been downloaded so in this case the ration will increase but the torrent  not showing in completed torrents in tracker , but if you implement a complete event it will hard to tracker to know

**Comment by flightcrank** [comment]
> upload kb amount only works with arrows not increasing when we typing desired amount of kb.

yes it does work. type in amount with keyboard and press enter. this is the default behaviour of java's swing JSpinner object.

>if i want to increase the ratio on random torrent not been downloaded so in this case the ratio will increase but the torrent wil not showing in completed torrents in tracker , but if you implement a complete event it will be hard to tracker to know

as stated before this is allowed most if not all private trackers. In the event of cross seeding from a different tracker. Also not every file in a torrent needs to be downloaded. for example Screenshots or .nfo's in a movie torrent never triggering a complete event. Its not evidence of cheating and as long as you use sane upload speeds the tracker will never know. 

No convincing reason has been for its inclusion However the source is open so feel free to push a commit with this feature added, and ill accept it in.

**Comment by flightcrank** [comment]
> upload kb amount only works with arrows not increasing when we typing desired amount of kb.

yes it does work. type in amount with keyboard and press enter. this is the default behaviour of java's swing JSpinner object.
> if i want to increase the ratio on random torrent not been downloaded so in this case the ratio will increase but the torrent wil not showing in completed torrents in tracker , but if you implement a complete event it will be hard to tracker to know

as stated before this is allowed most if not all private trackers. In the event of cross seeding from a different tracker. Also not every file in a torrent needs to be downloaded. for example Screenshots or .nfo's in a movie torrent never triggering a complete event. Its not evidence of cheating and as long as you use sane upload speeds the tracker will never know.

No convincing reason has been for its inclusion However the source is open so feel free to push a commit with this feature added, and ill accept it in.
